// Steam licensecache Protobuf schema.
//
// Defines the format of the decrypted licensecache file stored at
// userdata/{steam32_id}/config/licensecache.
//
// Source: SteamDatabase/Protobufs (steammessages_clientserver.proto)
// Used by: Depressurizer (SteamLicenseParser.cs)
//
// Regenerate with: protoc --python_out=. licensecache.proto

syntax = "proto2";

message CMsgClientLicenseList {
    optional int32 eresult = 1 [default = 2];
    repeated License licenses = 2;

    message License {
        optional uint32 package_id = 1;
        optional uint32 time_created = 2;
        optional uint32 time_next_process = 3;
        optional int32 minute_limit = 4;
        optional int32 minutes_used = 5;
        optional uint32 payment_method = 6;
        optional uint32 flags = 7;
        optional string purchase_country_code = 8;
        optional uint32 license_type = 9;
        optional int32 territory_code = 10;
        optional int32 change_number = 11;
        optional uint32 owner_id = 12;
        optional uint32 initial_period = 13;
        optional uint32 initial_time_unit = 14;
        optional uint32 renewal_period = 15;
        optional uint32 renewal_time_unit = 16;
        optional uint64 access_token = 17;
        optional uint32 master_package_id = 18;
    }
}
